subprojects/freetype2.wrap

```diff
@@ -0,0 +1,8 @@
+[wrap-git]
+directory = freetype2
+url = https://gitlab.freedesktop.org/freetype/freetype.git
```

Comment:
freetype2, fribidy and harfbuzz are all available in WrapDB. You  can use `meson wrap install <name>` to fetch their .wrap file. That will pin them on stable releases which makes your CI often more stable. Unless there is a need for tracking git master?

Reply:
I believe there was a desire to not have to introduce periodic commits bumping them. I have no strong opinion here myself and defer to the maintainers.

Reply:
Meson flattens the subproject namespace and builds all subprojects via the top-level project's own subprojects/ directory, regardless of which project actually triggers building the subproject.

If Aegisub has a wrap file for zlib/libpng/freetype2/whatever, then what libass has or doesn't have is totally irrelevant, since meson will not open the file for reading nor use it to download and/or configure a subproject.

If Aegisub does NOT have a given wrap file, but libass does, then meson can *hoist* it into the top-level subprojects/ directory for Aegisub itself, and then perform the actual tarball / git download there as well.

Reply:
One possibility is to do this:

```
meson wrap update-db
```

meson will download a `subprojects/wrapdb.json` and can then later use this to do automatic fallback even when a .wrap file does not exist. Be warned: this is a bit YOLO as you implicitly trust that whatever your https://wrapbd.mesonbuild.com https connection returns is trustworthy for doing code execution. It will download .wrap files with cryptographic hashes in them, but nothing guarantees the .wrap file is trustworthy. Most likely, your connection won't be hijacked due to the use of HTTPS. Also, most likely, no wrapdb maintainer will go rogue and start changing wrap files around to point to malicious urls.

I would say it's perfectly safe to use in CI. I wouldn't personally use it for locally building and running random applications, though this is all very conditional on your trust model.

Reply:
@torque wrap files are still there; instead just like for e.g. `zlib` already  CI runs which use it should install it from wrapdb